components/CandidateList/CandidateList.tsx

```diff
@@ -1,27 +1,28 @@
-﻿import React from "react";
+﻿import React, {useEffect, useState} from "react";
 import scss from './CandidateList.module.scss';
-import {getCandidates} from "../../api/candidatesApiClient";
+import {Candidate, getCandidates, ListResponse} from "../../api/candidatesApiClient";
 import {CandidateLink} from "../CandidateLink/CandidateLink";
 
+interface CandidatesListProps {
+    fetchCandidates: () => Promise<ListResponse<Candidate>>;
+}
 
-export function CandidateList(): JSX.Element {
+export function CandidateList(props: CandidatesListProps): JSX.Element {
 
     //TODO: use getCandidates function to get actual data
-    
-    type Candidate = {
-        name: string;
-        id: number;
-    }
-    
-    const candidates: Candidate[] = [
-        {name: "Michael Walker", id: 1}, 
-        {name: "Stephen Jackson", id: 2},
-        ];
+    const [candidates, setCandidates] = useState<Candidate[]>([]);
+
+    useEffect(() => {
+        props.fetchCandidates()
+            .then(response => setCandidates(response.items));
+    }, [props]);
```

Comment:
If we put this code here, then we will

- run it once on the server side, where Next generates the server side HTML (i expect this will fail because it doesn't know what `fetch` is??
- run it again on the client side, where Next re-runs all the react code.

So we are going to end up making more requests to the server than we really need to.

Next has some nice approaches for preventing this.
Take a look at this - https://nextjs.org/docs/basic-features/data-fetching#getserversideprops-server-side-rendering

this will cause the code to be run once on the server, but then _not_ be run again client side, which is _probably_ what we'd like in this case.